STYLE.md

```diff
@@ -0,0 +1,453 @@
+That's why *100* is a reasonable compromise.
+
+
+### Indentation
+
+* Do not vertically align code. Mainly because the need to change identifiers and re-aligning the other lines causes semantically null changes in the diff.
+* Tabs are illegal.
+* Use spaces for indenting.
+* Indent your code blocks with *2 spaces*.
+* Indent the `where` keyword with one space to set it
+apart from the rest of the code and indent the definitions in a
+`where` clause with 1 space.
+
+Some examples:
+
+```haskell
+sayHello :: IO ()
+sayHello = do
+  name <- getLine
+  putStrLn $ greeting name
+ where
+  greeting :: String -> String
+  greeting name = "Hello, " ++ name ++ "!"
+
+filter :: (a -> Bool) -> [a] -> [a]
+filter _ [] = []
+filter p (x:xs)
+  | p x = x : filter p xs
+  | otherwise = filter p xs
+```
+
+### Blank Lines
+
+* One blank line between top-level definitions.
+* No blank lines between type signatures and function definitions.
+* Add one blank line between
+functions in a type class instance declaration if the function bodies
+are large.
+* You can add blank lines inside a big `do` block to separate logical
+parts of it.
+* You can also use blank lines to separate definitions inside `where`
+clause, if necessary.
+
+### Whitespace
+
+* Surround binary operators with a single space on either side. In case of
+currying add one space between the argument and the operation.
+
+* Use some tool to remove trailing spaces.
+
+### Naming convention
+
+**Casing**:
+
++ **_lowerCamelCase_** for function and variable names.
++ **_UpperCamelCase_** for types.
+
+* Don't use short names like `n`, `sk`, `f` unless their meaning is clear from
+context (function name, types, other variables, etc.).
+
+* For readability reasons, don't capitalize all letters when using an
+abbreviation.  For example, write `HttpServer` instead of
+`HTTPServer`.  Exception: two or three letter abbreviations, e.g. `IO`, `STM`.
+
+**Records name conventions**
+
+If data type has only one constructor then this data type name should be the same
+as constructor name (also applies to `newtype`).
+
+```haskell
+data User = User Int String
+```
+
+**Field names for record data types**
+
+Choose a reasonable prefix is used such that we can ensure the field name will remain unique throughout the package. Use your discretion.
+
+e.g
+```haskell
+data MempoolEnv m blk = MempoolEnv
+  { meLedger :: LedgerInterface m blk
+  , meLedgerCfg :: LedgerConfig blk
+  , meStateVar :: StrictTVar m (InternalState blk)
+  , meTracer :: Tracer m (TraceEventMempool blk)
+  }
+```
+
+### Comments
+
+#### Punctuation
+
+Write proper sentences; start with a capital letter and use proper
+punctuation.
+
+#### Top-Level Definitions
+
+* Comment top level functions (particularly exported functions)
+as much as possible. If it is not blindingly obvious what a function
+is doing, you should add a comment.
+* Always provide a type signature
+* Use Haddock syntax in the comments.
+* Comment every exported data type.
```

Comment:
Even the criteria of blinding obviousness applies here as well?  Say a newtype around a `FilePath`.

If it literally means a `data` ADT, then we perhaps want to clarify that, @Jimbo4350.

Reply:
Yeah, I do not comment newtype wrappers around `FilePath` and other obvious one.

This is an example of where hard rules like "all types need comments" don't make much sense.

Reply:
Will relax the wording here too.